BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -835,7 +792,25 @@ def generateFibulaPlanes(self):
       shNode.SetItemParent(dynamicModelerNodeItemID, planeCutsFolder)
       modelNodeItemID = shNode.GetItemByDataNode(modelNode)
       shNode.SetItemParent(modelNodeItemID, cutBonesFolder)
+  
+  def getRotationMatrixFromAxis1ToAxis2(self,axis1,axis2):
+    rotationMatrix = vtk.vtkMatrix4x4()
+    rotationMatrix.DeepCopy((1, 0, 0, 0,
+                             0, 1, 0, 0,
+                             0, 0, 1, 0,
+                             0, 0, 0, 1))
+    #range(3) because rotation matrix is 3x3, the forth colomn and forth row are unchanged
+    for i in range(3):
+      for j in range(3):
+        projection = [0,0,0]
+        vtk.vtkMath.ProjectVector(axis2[i],axis1[j],projection)
+        if vtk.vtkMath.Dot(projection,axis1[j]) >= 0:
+          rotationMatrix.SetElement(i,j,vtk.vtkMath.Norm(projection))
+        else:
+          rotationMatrix.SetElement(i,j,-vtk.vtkMath.Norm(projection))

```

Comment:
Rotation transforms are wrong, it's visible on the 3DView. I've checked the function formula and I can't figure out why it doesn't work right.

Reply:
If two coordinate systems are defined by their transform to world (or some other common reference coordinate system) then you can get relative transforms between them by multiplying one with the inverse of the other:

    AToB = inv(BToWorld) * AToWorld

I assumed here that "axis1" is a transformation, which would mean that the names are extremely misleading. Names and types must be absolutely clear, especially at method inputs/outputs. Name of a variable must be ...Vector, ...Point, ...To...Transform. If you work with multiple coordinate systems then you also need to add the coordinate system name to the name of the variable: ...Vector_A, ...Point_A (means their coordinates are defined in coordinate system A).

I would also recommend to use the same type for storing a transformation in a member variable or passing it as function input/output. It can be either vtkMatrix4x4 or 4x4 numpy array, but it should be always the same (you can convert using slicer.util helper functions). Otherwise you will need to keep guessing if you pass the correct type.

Reply:
There was a small mistake in the _SetElement_ call, instead of:
`rotationMatrix.SetElement(i,j,vtk.vtkMath.Norm(projection))`
it was:
`rotationMatrix.SetElement(j,i,vtk.vtkMath.Norm(projection))`
Now it works well.
I'll change the name of the arguments of the function to something more meaningfull and I'll change all the argument types to vtkMatrix4x4. This mistake could be avoided if I set up well the argument types